scripts/check_no_std.sh

```diff
@@ -1,18 +1,21 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
-target=riscv32imac-unknown-none-elf
+target=wasm32-unknown-unknown
```

Comment:
wasm32 should be even more restrictive than the riscv target so this makes sense

Reply:
It's not, wasm32 has std

Reply:
Should there be a separate check_wasm.sh script?

Reply:
You can update the workflow `cargo hack` to also check for wasm feature https://github.com/alloy-rs/alloy/blob/f37ac3051c46248542d350c33f8c529fc59e8de3/.github/workflows/ci.yml#L92

Reply:
Okay I moved the build check to ci.yml